client/order/index.js

```diff
@@ -108,6 +110,20 @@ jQuery( function ( $ ) {
 			this.value === 'wc-cancelled' &&
 			originalStatus !== 'wc-cancelled'
 		) {
+			// If order has an uncaptured authorization, confirm
+			// that merchant indeed wants to cancel both the order
+			// and the authorization.
+			if ( hasOpenAuthorization ) {
+				renderModal(
+					<CancelAuthorizationConfirmationModal
+						originalOrderStatus={ originalStatus }
+					/>
+				);
+				return;
+			}
+			// If it is possible to refund an order, double check that
+			// merchants indeed wants to cancel, or if they just want to
+			// refund.
```

Comment:
❤️ the comments. Not related to your change, but it make sense to keep consistency and refactor conditional for the refund confirmation:

```
if ( canRefund && refundAmount > 0) {
renderModal(
		<CancelConfirmationModal
			originalOrderStatus={ originalStatus }
		/>
);
}
```

Reply:
Done in 087c3ceac